.github/workflows/icu_merge_ci.yml

```diff
@@ -36,3 +36,163 @@ jobs:
+
+  # Compile libraries used by all ICU4C performance tests.
+  icu4c-store-perf-libs:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Build ICU libraries
+        run: |
+          cd icu4c/source;
+          ./runConfigureICU Linux;
+          make -j 2 all;
+          ls -l;
+          cd lib;
+          cp -Ps ../tools/ctestfw/libicutest* .
+          ls -l;
+      - name: Upload ICU libraries
+        uses: actions/upload-artifact@v2
+        with:
+          name: icu-perf-libs
+          path: icu4c/source/lib
+          retention-days: 1
+
+  # Assorted ICU4C performance tests
+  icu4c-performance-tests:
+    needs: icu4c-store-perf-libs
+    strategy:
+      fail-fast: false
+      matrix:
+        perf: [charperf, usetperf, utfperf]
+        include:
+          - perf: charperf
+            tests: TestIsAlpha TestIsUpper TestIsLower TestIsDigit TestIsSpace TestIsAlphaNumeric TestIsPrint TestIsControl TestToLower TestToUpper TestIsWhiteSpace
+          - perf: usetperf
+            tests: titlecase_letter_add  titlecase_letter_contains titlecase_letter_iterator unassigned_add unassigned_contains unassigned_iterator pattern1 pattern2 pattern3
+          - perf: utfperf
+            tests: Roundtrip FromUnicode FromUTF8
+            file: "-f ../../icu4j/perf-tests/data/conversion/xuzhimo.txt"
+            flag: "-e gb18030"
+
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Create directory for lib files
+        run: mkdir icu4c/source/perflib
+      - name: Get ICU libs
+        uses: actions/download-artifact@v2
+        with:
+          name: icu-perf-libs
+          path: icu4c/source/lib
+
+      - name: Build ICU4C incl. perf tests
+        run: |
+          cd icu4c/source;
+          ./runConfigureICU Linux;
+          (cd test/perf && make -j2 everything && mkdir -p results && mkdir -p results/${{ matrix.perf }})
+          LD_LIBRARY_PATH=lib ./test/perf/${{ matrix.perf }}/${{ matrix.perf }} ${{ matrix.flag }} -t 5 -p 10 ${{ matrix.file }} ${{ matrix.tests }} | tee test/perf/results/${{ matrix.perf }}/output.txt
+
+      - name: Store performance test results
+        uses: gregtatum/github-action-benchmark@d3f06f738e9612988d575db23fae5ca0008d3d12
+        with:
+          # The perf tests result data is in ndjson format.
+          tool: 'ndjson'
+          output-file-path: icu4c/source/test/perf/results/${{ matrix.perf }}/output.txt
+          # Tentative setting.
+          alert-threshold: '100%'
+          fail-on-alert: true
+          gh-pages-branch: perfdata
+          benchmark-data-dir-path: perf/results/${{ matrix.perf }}
+          auto-push: true
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          comment-on-alert: true
+
+  # Assorted ICU4C performance tests using data files
+  icu4c-performance-tests-with-files:
+    needs: icu4c-store-perf-libs
+    strategy:
+      fail-fast: false
+      matrix:
```

Comment:
When you have functionality that allows include lists and exclude lists (ex: GHA; ex: rsync), it is good to document the behavior of which list is applied before the other.

Maybe make a note based on the info from the [GHA docs on matrix jobs](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions)?:
```
Note: All include combinations are processed after exclude. This allows you to use include to add back combinations that were previously excluded.
```

Reply:
Comment added.